src/azure-cli/azure/cli/command_modules/cloud/custom.py

```diff
@@ -123,8 +127,8 @@ def register_cloud(cmd,
                    suffix_azure_datalake_store_file_system_endpoint=None,
                    suffix_azure_datalake_analytics_catalog_and_job_endpoint=None,
                    suffix_acr_login_server_endpoint=None):
-    c = _build_cloud(cmd.cli_ctx, cloud_name, cloud_config=cloud_config,
-                     cloud_args=locals())
+    c = _build_cloud(cmd.cli_ctx, cloud_name, skip_endpoint_discovery=skip_endpoint_discovery,
+                     cloud_config=cloud_config, cloud_args=locals())
```

Comment:
Using `locals()` is very hacky and unpredictable. It may end up with unwanted variables being recorded in the cloud.

Reply:
I know. Technical debt, not necessarily to address right now.